google/cloud/security/common/util/log_util.py

```diff
@@ -12,29 +12,90 @@
+    that Forseti is running on Compute Engine.
+  - Otherwise, fall back to local logging.
+"""
 
 import logging
-import os
-
-class LogUtil(object):
-    """Utility to wrap logging setup."""
-
-    @classmethod
-    def setup_logging(cls, module_name):
-        """Setup logging configuration.
-
-        Args:
-            module_name: The name of the module to describe the log entry.
-        """
-        formatter = logging.Formatter(
-            '%(asctime)s %(name)-12s %(levelname)-8s %(message)s')
-        handler = logging.StreamHandler()
-        handler.setFormatter(formatter)
-        my_logger = logging.getLogger(module_name)
-        my_logger.addHandler(handler)
-        if os.getenv('DEBUG'):
-            my_logger.setLevel(logging.DEBUG)
-        else:
-            my_logger.setLevel(logging.INFO)
-        return my_logger
+
+import gflags as flags
+
+from google.cloud.security.common.gcp_api import cloud_logging
+from google.cloud.security.common.gcp_api import compute
+
+FLAGS = flags.FLAGS
+flags.DEFINE_boolean('use_cloud_logging', False,
+                     'Use Cloud Logging, if available.')
+flags.DEFINE_boolean('nouse_cloud_logging', False, 'Do not use Cloud Logging.')
+flags.DEFINE_boolean('debug', False, 'Show DEBUG level logs.')
+
+LOG_FORMAT = '%(asctime)s %(name)-12s %(levelname)-8s %(message)s'
+MODULE_PREFIX = 'google.cloud.security'
+
+def get_logger(module_name):
+    """Setup the basic logger.
+
+    Args:
+        module_name: The name of the mdule to describe the log entry.
+
+    Returns:
+        An instance of the configured logger.
+    """
+    formatter = logging.Formatter(LOG_FORMAT)
+    console_handler = logging.StreamHandler()
+    console_handler.setFormatter(formatter)
+    logger_instance = logging.getLogger(module_name)
+    logger_instance.addHandler(console_handler)
+    logger_instance.setLevel(logging.INFO)
+
+    return logger_instance
+
+def configure_logger(module_name):
+    """Additional logging configuration based on flags.
+
+    Args:
+        module_name: The name of the module for which to configure logging.
+    """
+    (is_gce, _) = compute.is_compute_engine_instance()
+    should_use_cloud_logger = None
+    cloud_handler = None
+
+    # Use Cloud Logging if flag is set or if on GCE instance.
+    if FLAGS.use_cloud_logging:
+        should_use_cloud_logger = True
+
+    # Force local logger.
+    if FLAGS.nouse_cloud_logging:
+        should_use_cloud_logger = False
+
+    # Determine whether to use cloud logger based on environment.
+    if should_use_cloud_logger is None:
+        should_use_cloud_logger = is_gce
+
+    if should_use_cloud_logger:
+        formatter = logging.Formatter(LOG_FORMAT)
+        cloud_handler = cloud_logging.CloudLoggingHandler(_get_cloud_logger())
+        cloud_handler.setFormatter(formatter)
+
+        logger_instance = logging.getLogger(module_name)
+        logger_instance.addHandler(cloud_handler)
+
+        if FLAGS.debug:
```

Comment:
you can't use flags until they are parsed, which happens in app.run(). So the strategy is to attach the default log handler at the toplevel (e.g. module import) and then attach the stackdriver handler after app.run().